source/standards/source-code/working-with-git.html.md.erb

```diff
@@ -142,37 +142,24 @@ commands like git merge and git revert.
 
 ## Branching/merging conventions
 
-You may often choose to work on a particular feature on a "feature branch"
-rather than directly on `main`. Indeed, given how cheap branches are in Git,
-[using branches](http://git-scm.com/book/en/Git-Branching-Basic-Branching-and-Merging) is positively encouraged.
-
-You are encouraged to make liberal use of Git's [history rewriting
-features](http://git-scm.com/book/en/Git-Tools-Rewriting-History) while working
-locally, in order to arrange your commits into appropriate logical chunks that
-will make sense to your fellow developers. In particular, you may find
-`git rebase --interactive` very useful. You are also encouraged to avoid merge
```

Comment:
I think here the advice to avoid merge commits when updating your feature branch with the lastest main is still good advice and worth keeping, and kind of separate to the 'do you merge or rebase when merging into main' debate.

Reply:
Maybe it can be clarified to something like

"If you are working on a feature branch and main gets updated, you may need to update your feature branch with main. In this case, it's useful to `git rebase main` rather than creating a merge commit on your feature branch, since this keeps the history cleaner."